packages/grid/x-data-grid-pro/src/tests/columnsVisibility.DataGridPro.test.tsx

```diff
@@ -0,0 +1,157 @@
+import * as React from 'react';
+import { spy } from 'sinon';
+import { expect } from 'chai';
+import { createRenderer } from '@mui/monorepo/test/utils';
+import {
+  DataGridPro,
+  GridApiRef,
+  useGridApiRef,
+  DataGridProProps,
+  GridRowsProp,
+  GridColumns,
+} from '@mui/x-data-grid-pro';
+
+const isJSDOM = /jsdom/.test(window.navigator.userAgent);
+
+const rows: GridRowsProp = [{ id: 1 }];
+
+const columns: GridColumns = [{ field: 'id' }];
+
+/**
+ * TODO: Remove deprecated tests on v6
+ */
+describe('<DataGridPro /> - Columns Visibility', () => {
+  const { render } = createRenderer();
+
+  let apiRef: GridApiRef;
+
+  const TestDataGridPro = (
+    props: Omit<DataGridProProps, 'columns' | 'rows' | 'apiRef'> &
+      Partial<Pick<DataGridProProps, 'rows' | 'columns'>>,
+  ) => {
+    apiRef = useGridApiRef();
+
+    return (
+      <div style={{ width: 300, height: 300 }}>
+        <DataGridPro
+          columns={columns}
+          rows={rows}
+          {...props}
+          apiRef={apiRef}
+          autoHeight={isJSDOM}
+        />
+      </div>
+    );
+  };
+
+  describe('apiRef: setColumnVisibility', () => {
+    describe('Model on 1st render: use not use `GridColDef.hide`', () => {
+      it('should update `visibleColumnsModel` but not `GridColDef.hide` in state', () => {
+        render(<TestDataGridPro initialState={{ columns: { visibleColumnsModel: ['id'] } }} />);
+        apiRef.current.setColumnVisibility('id', false);
+        expect(apiRef.current.state.columns.lookup.id.hide).to.equal(false);
```

Comment:
I can listen to `columnChange` events to get the same result if we really want to avoid state assertions.

Reply:
You can also check if `columnVisibilityChange` was not called. My point is that we don't recommend direct state access, but to use selectors or API methods instead. That way, the test should ensure that if someone access the state via one of the recommended ways it will return the correct value. We shouldn't be testing the internal implementation, but what is one the public API or can be seen (UI).

Reply:
Then I'll use the selector.
It won't impact the end user but users that use the selectors could have problems if Iweupdate the `hide` field by mistake.